protobuf/lib/src/protobuf/field_set.dart

```diff
@@ -83,9 +83,9 @@ class _FieldSet {
     return List.filled(length, null, growable: false);
   }
 
-  // Use a fixed length list and not a constant list to ensure that _values
-  // always has the same implementation type.
-  static final List _zeroList = [];
+  // Use `List.filled` and not a `[]` to ensure that `_values` always has the
+  // same implementation type.
+  static final List _zeroList = List.filled(0, null, growable: false);
```

Comment:
FYI

`List.empty()` is also fixed length.
`const []` would be fine here too, since it is also fixed length and AOT and dart2js both can understand mixed const and fixed-length lists.

Reply:
unless I am mistaken `const []` will be an instance of `_ImmutableArray` on the VM rather than an instance of a mutable array.

Reply:
Yes, but `_List` and `_ImmutableList` now have a common supertype `_Array` which holds most of the methods, allowing e.g. `operator []` to be a single method for both.

There is nothing wrong with the merged code, as using `List.filled(..., growable: false)` in both places makes it obvious they are the same internal type.

I'm not sure it is worth special-casing the empty list. There are not many proto mesages with no non-extension fields.